core/client.go

```diff
@@ -104,7 +104,12 @@ func (c *client) Publish(message *messages.Publish, data []byte) {
 	if debug {
 		log.Printf("Client from %s publishing %s to %s", c.GetRemoteAddr(), message.Content, message.Channel)
 	}
-
+	// Don't publish to a channel that client is not subscribed
```

Comment:
Remove, there is no need for a client to subscribe to a channel before publishing to it.